packages/graph-engine/src/nodes/curves/sampleFloatCurve.ts

```diff
@@ -0,0 +1,76 @@
+import { FloatCurve, INodeDefinition, ToInput, ToOutput } from '../../index.js';
+import { FloatCurveSchema, NumberSchema } from '../../schemas/index.js';
+import { Node } from '../../programmatic/node.js';
+import { arrayOf } from '../../schemas/utils.js';
+import { cubicBezier } from './floatCurve.js';
+import { setToPrecision } from '../../utils/precision.js';
+
+export default class NodeDefinition extends Node {
+	static title = 'Sample Array from Float Curve';
+	static type = 'studio.tokens.curve.sampleFloatCurve';
+	static description = 'Samples a float curve at regular intervals';
+
+	declare inputs: ToInput<{
+		curve: FloatCurve;
```

Comment:
Why don't you just allow someone to generate the sample points? Then they can have better control over whether they want to sample densely within 0.4->0.8 for example